lib/assert.js

```diff
@@ -269,9 +269,7 @@ function _deepEqual(actual, expected, strict, memos) {
 
   const actualPosition = memos.actual.map.get(actual);
   if (actualPosition !== undefined) {
-    if (actualPosition === memos.expected.map.get(expected)) {
-      return true;
-    }
+    return actualPosition === memos.expected.map.get(expected);
```

Comment:
Thank you for the point. I tried it and indeed, it makes the case pass. However, at first, it breaks (or changes expectations of the) test added by the PR from "throws" to "not throws", which is probably not a bad thing, but rather a point that requires some discussion. Secondly, I was able to create a case that still fails:
```javascript
var a = [ 1, 2 ]
var b = [ 3, 4 ]
var c = [ 1, 2 ]
var d = [ 3, 4 ]

assertDeepAndStrictEqual({
  v: a,
  s: new Set([a, b])
}, {
  v: c,
  s: new Set([d, c])
});
```